dali/test/python/nose_utils.py

```diff
@@ -79,18 +81,18 @@ def assert_raises(exception, *args, glob=None, regex=None, match_case=None, **kw
     """
 
     if glob is None and regex is None:
-        return tools.assert_raises(exception, *args, **kwargs)
+        return unittest.TestCase.assertRaise(exception, *args, **kwargs)
```

Comment:
Just wanted to get rid of nose dependency. I can revert if you feel that this is not that simple (I cannot test this in CI as nvImageCodecs doesn't work for Python 3.12 yet and DALI depends on it).

Reply:
Maybe it's better to revisit it separately - if it makes sense to do the same trick independently from nose. I am quite convinced that it would not work just like that.

Reply:
Reverted